File: src/utils/formatters.ts
// Formatear monto a entero (las APIs chilenas usan montos sin decimales)
export function formatAmount(amount: number): number {
  return Math.round(amount);
}

// Formatear fecha a ISO string
export function formatDate(date: Date): string {
  return date.toISOString();
}

// Parsear fecha desde string
export function parseDate(dateString: string): Date {
  return new Date(dateString);
}

// Generar ID único simple
export function generateId(): string {
  return `${Date.now()}-${Math.random().toString(36).substr(2, 9)}`;
}

// Limpiar objeto de valores undefined/null
export function cleanObject<T extends Record<string, any>>(obj: T): Partial<T> {
  return Object.entries(obj).reduce((acc, [key, value]) => {
    if (value !== undefined && value !== null) {
      acc[key] = value;
    }
    return acc;
  }, {} as any);
}

File: kotlin/src/main/kotlin/cl/payments/utils/Formatters.kt
package cl.payments.utils

import java.time.LocalDateTime
import java.time.format.DateTimeFormatter

// Formatear monto a entero (las APIs chilenas usan montos sin decimales)
fun formatAmount(amount: Int): Int {
    return amount
}

// Formatear fecha a ISO string
fun formatDate(date: LocalDateTime): String {
    return date.format(DateTimeFormatter.ISO_DATE_TIME)
}

// Parsear fecha desde string
fun parseDate(dateString: String): LocalDateTime {
    return LocalDateTime.parse(dateString, DateTimeFormatter.ISO_DATE_TIME)
}

// Generar ID único simple
fun generateId(): String {
    return "${System.currentTimeMillis()}-${(0..999999).random()}"
}

// Limpiar objeto de valores null
fun <K, V> cleanMap(map: Map<K, V?>): Map<K, V> {
    return map.filterValues { it != null }.mapValues { it.value!! }
}